src/app/services/actions/adaptive-cards-action-creator.ts

```diff
@@ -0,0 +1,82 @@
+import { IAction } from '../../../types/action';
+import { IQuery } from '../../../types/query-runner';
+import { ADAPTIVE_FETCH_ERROR , ADAPTIVE_FETCH_PENDING, ADAPTIVE_FETCH_SUCCESS } from '../redux-constants';
+
+export function getAdaptiveCardSuccess(result: string): IAction {
+  return {
+    type: ADAPTIVE_FETCH_SUCCESS,
+    response: result,
+  };
+}
+
+export function getAdaptiveCardError(error: string): IAction {
+  return {
+    type: ADAPTIVE_FETCH_ERROR,
+    response: error
+  };
+}
+
+export function getAdaptiveCardPending(): IAction {
+  return {
+    type: ADAPTIVE_FETCH_PENDING,
+    response: ''
+  };
+}
+
+export function getAdaptiveCard(payload: string, sampleQuery: IQuery): Function {
+  return async (dispatch: Function) => {
+
+    if (!payload) {
+      // no payload so return empty result
+      return dispatch(getAdaptiveCardSuccess(''));
```

Comment:
Instead of calling `getAdaptiveCardSuccess` with an empty string as an argument, you can take advantage of default parameters.

```js
function getAdaptiveCardSuccess(result = '') {}
```

Then when there's not payload you just call `getAdaptiveCardSuccess()`

Reply:
+1